pyccel/codegen/printing/fcode.py

```diff
@@ -2000,11 +2000,13 @@ def _print_Is(self, expr):
         if isinstance(expr.rhs, Nil):
             return '.not. present({})'.format(lhs)
 
-        if a.dtype is NativeBool() and b.dtype is NativeBool():
+        if (a.dtype is NativeBool() and b.dtype is NativeBool()):
             return '{} .eqv. {}'.format(lhs, rhs)
 
-        errors.report(PYCCEL_RESTRICTION_IS_RHS, symbol=expr,
-            severity='fatal')
+        if (a.dtype is NativeInteger() and b.dtype is NativeInteger()):
+            return '{} .eq. {}'.format(lhs, rhs)
+
+        raise NotImplementedError("Only booleans, integers and None are currently supported for comparison")
```

Comment:
Why do we use a `NotImplementedError` instead of calling `errors.report`? Can't we use the message stored in `PYCCEL_RESTRICTION_IS_RHS`? If we want to keep this message then it should be modified to indicate that these types are only supported *for the is operator*, these restrictions do not exist for comparison operators in general